src/webgpu/api/validation/createView.spec.ts

```diff
@@ -1,311 +1,318 @@
-    } = {}
-  ): GPUTextureViewDescriptor {
-    const {
-      format = FORMAT,
-      dimension = '2d',
-      baseMipLevel = 0,
-      mipLevelCount = MIP_LEVEL_COUNT,
-      baseArrayLayer = 0,
-      arrayLayerCount = 1,
-    } = options;
-    return {
-      format,
-      dimension,
-      baseMipLevel,
-      mipLevelCount,
-      baseArrayLayer,
-      arrayLayerCount,
-    };
-  }
-}
 
-export const g = makeTestGroup(F);
+    const success = textureFormat === viewFormat;
+    t.expectValidationError(() => {
+      texture.createView({ format: viewFormat });
+    }, !success);
+  });
 
-g.test('creating_texture_view_on_a_2D_non_array_texture')
+g.test('dimension')
   .desc(
-    `TODO: write description, and shorten name
-
-  TODO: mipLevelCount == 0 should mean 0, not "auto". "undefined" means "auto".`
+    `For all {texture dimension}, {view dimension}, test that they must be compatible:
+  - 1d -> 1d
+  - 2d -> 2d, 2d-array, cube, or cube-array
+  - 3d -> 3d`
   )
-  .paramsSimple([
-    { _success: true }, // default view works
-    { arrayLayerCount: 1, _success: true }, // it is OK to create a 2D texture view on a 2D texture
-    { arrayLayerCount: 2, _success: false }, // it is an error to view a layer past the end of the texture
-    { dimension: '2d-array' as const, arrayLayerCount: 1, _success: true }, // it is OK to create a 1-layer 2D array texture view on a 2D texture
-    // mip level is in range
-    { mipLevelCount: 1, baseMipLevel: MIP_LEVEL_COUNT - 1, _success: true },
-    { mipLevelCount: 2, baseMipLevel: MIP_LEVEL_COUNT - 2, _success: true },
-    // baseMipLevel == k && mipLevelCount == 0 means to use levels k..end
-    { mipLevelCount: 0, baseMipLevel: 0, _success: true },
-    { mipLevelCount: 0, baseMipLevel: 1, _success: true },
-    { mipLevelCount: 0, baseMipLevel: MIP_LEVEL_COUNT - 1, _success: true },
-    { mipLevelCount: 0, baseMipLevel: MIP_LEVEL_COUNT, _success: false },
-    // it is an error to make the mip level out of range
-    { mipLevelCount: MIP_LEVEL_COUNT + 1, baseMipLevel: 0, _success: false },
-    { mipLevelCount: MIP_LEVEL_COUNT, baseMipLevel: 1, _success: false },
-    { mipLevelCount: 2, baseMipLevel: MIP_LEVEL_COUNT - 1, _success: false },
-    { mipLevelCount: 1, baseMipLevel: MIP_LEVEL_COUNT, _success: false },
-  ])
-  .fn(async t => {
-    const { dimension = '2d', arrayLayerCount, mipLevelCount, baseMipLevel, _success } = t.params;
-
-    const texture = t.createTexture({ arrayLayerCount: 1 });
+  .params(u =>
+    u
+      .combine('textureDimension', kTextureDimensions)
+      .combine('viewDimension', kTextureViewDimensions)
+  )
+  .fn(t => {
+    const { textureDimension, viewDimension } = t.params;
 
-    const descriptor = t.getDescriptor({
-      dimension,
-      arrayLayerCount,
-      mipLevelCount,
-      baseMipLevel,
+    const texture = t.device.createTexture({
+      format: 'rgba8unorm',
+      dimension: textureDimension,
+      size: [4, 4, 6],
```

Comment:
`size` is incorrect for `1d`, which is not supported yet but we can make it correct in advance.